Tests/RudderStackAnalyticsTests/LifecycleManagementTests/LifecycleObserverTests.swift

```diff
@@ -6,98 +6,105 @@
+        lifecycleObserver?.addObserver(mockListener)
+        NotificationCenter.default.post(name: event.notificationName, object: nil)
+        
+        #expect(mockListener[keyPath: expectedFlag] == true, "Expected correct callback for \(event)")
+        
+        let allFlags: [KeyPath<MockLifecycleEventListener, Bool>] = [
+            \.onBackgroundCalled,
+            \.onForegroundCalled,
+            \.onTerminateCalled,
+            \.onBecomeActiveCalled
+        ]
+        
+        for flag in allFlags where flag != expectedFlag {
+            #expect(mockListener[keyPath: flag] == false, "Expected \(flag) to remain false for \(event)")
         }
+    }
+    
+    @Test("when multiple observers are registered, then all should receive lifecycle event callbacks")
+    func testMultipleObservers() {
+        let mockListener1 = MockLifecycleEventListener()
+        let mockListener2 = MockLifecycleEventListener()
+        
+        lifecycleObserver?.addObserver(mockListener1)
+        lifecycleObserver?.addObserver(mockListener2)
+        NotificationCenter.default.post(name: AppLifecycleEvent.background.notificationName, object: nil)
+        
+        #expect(mockListener1.onBackgroundCalled == true)
+        #expect(mockListener2.onBackgroundCalled == true)
+    }
+    
+    @Test("when sequential lifecycle events are posted, then they should be handled correctly")
+    func testSequentialEventHandling() {
+        lifecycleObserver?.addObserver(mockListener)
+        NotificationCenter.default.post(name: AppLifecycleEvent.foreground.notificationName, object: nil)
+        NotificationCenter.default.post(name: AppLifecycleEvent.becomeActive.notificationName, object: nil)
+        NotificationCenter.default.post(name: AppLifecycleEvent.background.notificationName, object: nil)
+        
+        #expect(mockListener.onForegroundCalled == true)
+        #expect(mockListener.onBecomeActiveCalled == true)
+        #expect(mockListener.onBackgroundCalled == true)
+    }
+    
+    @Test("when observer removal during event processing is attempted, it should work correctly")
+    func testObserverRemovalDuringEventProcessing() {
+        let mockListener1 = MockLifecycleEventListener()
+        let mockListener2 = MockLifecycleEventListener()
+        
+        lifecycleObserver?.addObserver(mockListener1)
+        lifecycleObserver?.addObserver(mockListener2)
+        
+        lifecycleObserver?.removeObserver(mockListener1)
+        NotificationCenter.default.post(name: AppLifecycleEvent.background.notificationName, object: nil)
+        
+        #expect(mockListener1.onBackgroundCalled == false)
+        #expect(mockListener2.onBackgroundCalled == true)
+    }
 }
 
-// MARK: - Mocks
+// MARK: - MockLifecycleEventListener
 
-final class LifecycleEventListenerMock: LifecycleEventListener {
+final class MockLifecycleEventListener: LifecycleEventListener {
```

Comment:
Let’s move the mock classes to a new file. We can also create a `Mock` folder to keep everything organised.

Reply:
Can you add a TODO for visibility?